TASK-54594 permit to managers, redactors, and super manager to edit events in space.

montassarhamadi opened the pull request:
Problem: Prior this change, if a space contains a redactor many problems is appeared .A member of space cannot edit events, whatever the role (super manager, redactor, manager) except of  event creator. On technical side,there is  incorrect logical expression returned by this method `canEditCalendar()`.
Fix: after fixing returned values of these two methods,the current behavior is:
-super manager , manager and redactors, can edit all events in space .
-the owner of event can always edit it
-toggle button "modify event" is enabled if a space has no redactor else it remains disabled for all

ahamdi commented on agenda-services/src/main/java/org/exoplatform/agenda/util/Utils.java:
```diff
@@ -389,7 +389,7 @@ public static boolean canInviteeEdit(IdentityManager identityManager,
     } else if (StringUtils.equals(SpaceIdentityProvider.NAME, requestedOwner.getProviderId())) {
       Space space = spaceService.getSpaceByPrettyName(requestedOwner.getRemoteId());
-      return space != null && (space.getRedactors() == null || space.getRedactors().length == 0);
+      return space != null && space.getRedactors() != null && space.getRedactors().length > 0;
```
Could space managers edit events ?

montassarhamadi replied:
> Could space managers edit events ?

In fact no, I have a problem when I remove a redactor and try to change the status of button toggle.But I don't understand what is redactor's income in enable or disable of this button ??!!

montassarhamadi committed:
TASK-54594 permit to  managers, redactors, and super manager to edit events in space.

Problem: Prior this change, if a space contains a redactor many problems is appeared .A member of space cannot edit events, whatever the role (super manager, redactor, manager) except of  event creator. On technical side,there is  incorrect logical expression returned by this method `canEditCalendar()`.
Fix: after fixing returned values of these two methods,the current behavior is:
-super manager , manager and redactors, can edit all events in space .
-the owner of event can always edit it
-toggle button "modify event" is enabled if a space has no redactor else it remains disabled for all

ahamdi commented on agenda-services/src/main/java/org/exoplatform/agenda/service/AgendaCalendarServiceImpl.java:
```diff
@@ -217,7 +218,7 @@ public Calendar createCalendarInstance(long ownerId) {
     boolean canEditCalendar = Utils.canEditCalendar(identityManager, spaceService, ownerId, userIdentityId);
     boolean canCreateEvent = Utils.canCreateEvent(identityManager, spaceService, ownerId, userIdentityId);
-    boolean canInviteeEdit = Utils.canInviteeEdit(identityManager, spaceService, ownerId);
+    boolean canInviteeEdit = Utils.canInviteeEdit(identityManager, spaceService, ownerId, RestUtils.getCurrentUser());
```
The toggle for enabling an invitee to edit if he is member of a space that does not have redactors, no need to add the user to this function.
The logic is as follows :
 - If there is at least a redactor in the space, the toggle of invite can Edit will be disabled and can not be enabled
 - If there is no redactor, the toggle is enabled and we can choose to let event invitees to edit the event
 - All managers and redactors in both cases are able to edit events

montassarhamadi replied:
How can I verify that this member is not a redactor if there is redactors in space to decide whether  toggle button will be enabled or disabled?

ahamdi commented on agenda-services/src/main/java/org/exoplatform/agenda/util/Utils.java:
```diff
@@ -424,7 +424,8 @@ public static boolean canEditCalendar(IdentityManager identityManager,
       boolean isRedactor = space != null && spaceService.isRedactor(space, userIdentity.getRemoteId());
       boolean spaceHasARedactor = space != null && space.getRedactors() != null && space.getRedactors().length > 0;
-      return (!spaceHasARedactor || isRedactor) && (superManager || isManager);
+      boolean spaceHasAManager = space != null && space.getManagers() != null && space.getManagers().length > 0;
+      return (spaceHasARedactor || spaceHasAManager) && (superManager || isManager || isRedactor);
```
```suggestion
      return (spaceHasARedactor && isRedactor) || superManager || isManager;
```
